### ISSUE TITLE
Parsing of internal variables

### ISSUE THREAD

**Issue by pjbisset** [bitfocus/companion-module-bmd-atem#137]
I am raising this request due to the closing comment in the same request in this issue: https://github.com/bitfocus/companion/issues/1248. There it was stated that parsing variables in an action is a per-module task. Also a later discussion: https://github.com/bitfocus/companion/issues/1508.

You provide the ability to set the recording filename. We have an ATEM Mini Pro and manually set the recording filename in the ATEM control software. I would like to be able to use something like the following to make up the filename value in Companion:

XXXX_$(internal:date_y)$(internal:date_m)$(internal:date_d)

This way I can press a button in Companion to update the recording filename automatically.

I can see at least one other issue (#120) that is similar, but based on the comment in the main issue above this has to be parsed per-action and my action is different to that raised in 120.

Thanks

**Comment by Julusian** [comment]
This is implemented for the recording filename action

**Comment by pjbisset** [comment]
I have just tested v2.1.4 of Companion and this functionality is unchanged - it still just writes the string of the variable name into the recording filename field in the ATEM software. Will this change be included in a future build of Companion or is there something I have to do to enable it? I have tried both Mac and Windows versions - same issue.

Also, when entering text for the button label it does not display the internal date options, only the time ones, in the popup list of available variables. Is this related to the module or Companion as a whole?
